Add info on specific build that is targeted (#17922)
docker/docs: build/building/multi-stage.md

$ docker build --target build -t hello .
```

> The builder targeted here is actually the first stage build responsible for building the Go application
> 
> ```dockerfile
> FROM golang:1.21 as build
> WORKDIR /src
> COPY <<EOF /src/main.go
> ```
> 

Reviewer: Thank you for your PR @kahenya-anita. Suggesting a small change to the wording here:

````suggestion
This build targets the stage responsible for building the Go application:

```dockerfile
FROM golang:1.21 as build
WORKDIR /src
COPY <<EOF /src/main.go
```

````

While reviewing the related issue, I noticed we have a typo on line 95 as well. It says the stage name is `builder`, but it should in fact be `build`. Could I ask you to please fix this typo as well? 

```
the stage named `builder`:
```

🙏🏻 thank you!
